src/types.rs

```diff
@@ -1530,6 +1560,19 @@ pub enum PersistentSubscriptionStatus {
     Live,
 }
 
+impl PersistentSubscriptionStatus {
+    pub(crate) fn from_string(str: String) -> Self {
+        match str.as_str() {
+            "NotReady" => PersistentSubscriptionStatus::NotReady,
+            "Behind" => PersistentSubscriptionStatus::Behind,
+            "OutstandingPageRequest" => PersistentSubscriptionStatus::OutstandingPageRequest,
+            "ReplayingParkedMessages" => PersistentSubscriptionStatus::ReplayingParkedMessages,
+            "Live" => PersistentSubscriptionStatus::Live,
+            _ => panic!("Unknown persistent subscription status '{}", str),
```

Comment:
Could there ever be a new persistent subscription status, or is that just never going to happen? Worth adding a `PersistentSubscriptionStatus::Unknown(String)`, to save the panic if there is one?

Reply:
Done